platform/android/os_android.cpp

```diff
@@ -313,8 +335,13 @@ bool OS_Android::main_loop_iterate() {
 }
 
 void OS_Android::main_loop_end() {
-	if (main_loop)
+	if (main_loop) {
+		if (main_loop->is_class("SceneTree")) {
+			SceneTree *scene_tree = Object::cast_to<SceneTree>(main_loop);
+			scene_tree->quit();
+		}
```

Comment:
Skip the unnecessary string check:
```suggestion
		SceneTree *scene_tree = Object::cast_to<SceneTree>(main_loop);
		if (scene_tree) {
			scene_tree->quit();
		}
```